pkg/generator/patch.go

```diff
@@ -0,0 +1,144 @@
+package generator
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	jsonpatch "github.com/evanphx/json-patch"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+	"sigs.k8s.io/kustomize/kyaml/yaml/merge2"
+	sigyaml "sigs.k8s.io/yaml"
+)
+
+// Apply strategic merge or json patch to the source YAML.
+func (gen *AssetGenerator) applyAssetPatch(sourceYAML []byte, patchAssetName string, extraReplacements []string) ([]byte, error) {
+	if strings.HasSuffix(patchAssetName, ".patch") {
+		return gen.applyJSONPatch(sourceYAML, patchAssetName, extraReplacements)
+	}
+	return gen.applyStrategicMergePatch(sourceYAML, patchAssetName, extraReplacements)
+}
+
+// Apply strategic merge patch to the source YAML.
+func (gen *AssetGenerator) applyStrategicMergePatch(sourceYAML []byte, patchAssetName string, extraReplacements []string) ([]byte, error) {
+	patchYAML := gen.mustReadAsset(patchAssetName, extraReplacements)
+	opts := yaml.MergeOptions{ListIncreaseDirection: yaml.MergeOptionsListAppend}
+	ret, err := merge2.MergeStrings(string(patchYAML), string(sourceYAML), false, opts)
+	if err != nil {
+		return nil, fmt.Errorf("failed to apply aseet %s: %v", patchAssetName, err)
```

Comment:
`%v` looks as a typo. Should we use `%w` everywhere, or there is a reason to prefer `%v` here?

Reply:
Also: `s/aseet/asset` :)

Reply:
Fixed.